File: 服务器/ChatServer/src/main/java/org/heima/chat/dao/InvitationDao.java
package org.heima.chat.dao;

import java.util.List;

import org.heima.chat.pojo.Invitation;

public interface InvitationDao
{
	void addInvitation(Invitation invitation);

	void updateInvitation(Invitation invitation);

	Invitation findInvitationBySender(String sender);

	List<Invitation> findInvitationByAccountAndState(String account, int state);
}

File: business/src/main/java/com/jw/business/db/dao/InvitationDao.kt
package com.jw.business.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update
import android.database.Cursor
import com.jw.business.db.model.Invitation

/**
 * 作者 : jinwangx
 * 创建时间 : 2018/6/13
 * 描述 : 数据库操作(邀请信息)
 */
@Dao
interface InvitationDao {

    @Query("select * from invitation where owner=:owner")
    fun getInvitationByOwner(owner: String): Cursor?

    @Query("select * from invitation where owner=:owner and invitator_account=:account limit 0,1")
    fun getInvitationByAccount(owner: String, account: String): Invitation

    @Query("select count(_id) from invitation where owner=:owner and agree=0")
    fun getUnAgreeCount(owner: String): Int

    @Insert
    fun addInvitation(invitation: Invitation):Long

    @Update
    fun updateInvitation(invitation: Invitation):Int
}